packages/SwingSet/src/vats/comms/clist-inbound.js

```diff
@@ -27,6 +27,45 @@ export function makeInbound(state, stateKit) {
 
   // *-LocalForRemote: receiving an object/promise from a remote machine
 
+  function retireRemotePromiseID(remoteID, rpid) {
+    insistRemoteType('promise', rpid);
+    const remote = getRemote(state, remoteID);
+    const lpid = remote.fromRemote.get(rpid);
+    assert(lpid, X`unknown remote ${remoteID} promise ${rpid}`);
+    assert(
+      remote.toRemote.has(lpid),
+      X`unmapped remote ${remoteID} promise ${lpid}`,
+    );
+    const p = state.promiseTable.get(lpid);
+    assert(
```

Comment:
If we're ready to retire this rpid, the promise must already be resolved, yeah? Should we assert that `p.state` is resolved instead of looking at the subscribers (which shouldn't exist at all, for resolved promises)?

Reply:
Ah, right, `retireRemotePromiseID` is doing double duty: it's called in `resolveFromRemote` to retire the clist entry for a promise that's about to be resolved (which means `p.state` is still unresolved, and the not-a-subscriber check is a good thing to do), and we should delete both directions of the clist.

It's also called after `resolveToRemote` has issued the resolution (and did `enqueueRemotePromiseIDRetirement`, and that remote has finalled acked the resolution, and so `retireAcknowledgedRemotePromiseIDs` is finishing the handshake. In this case, `p.state` should be resolved. I *think* that the outbound clist entry should have been deleted by now (see comment below), so in this case, we should only be deleting the inbound clist entry.

That might suggest that we should split this into two separate functions (or add a mode flag, eww).

Reply:
I think the subscribers check makes sense here since it's about the relationship between the rpid and the remote comms vat, which is the relationship we're fiddling with in this function.  A `p.state` check would not make sense here because that's about the local state of the promise, which is not part of the relationship with the remote.

BTW, the subscribers check (as well as various other bits) got in there by mirroring how we interact with the kernel, which does this.